docs/datamodel/links.rst

```diff
@@ -14,16 +14,259 @@ There are two kinds of link item declarations: *abstract links*, and
+from a source set is linked to only one object in the target set, and
+vice versa. For example, ``Employee`` and ``ReservedParking``:
+
+.. code-block:: sdl
+
+    type Employee {
+        required property name -> str;
+        single link parking -> ReservedParking {
+            constraint exclusive;
+        }
+    }
+    type ReservedParking {
+        required property number -> int64;
+    }
+
+An ``Employee`` can have up to one ``ReservedParking`` assigned
+exclusively to them. The :eql:constraint:`exclusive` constraint
+ensures that no more than *one* ``Employee`` can get the same
+``ReservedParking``, while the ``single`` qualifier on the link (which
+is the default, so it can be omitted) ensures that no ``Employee`` can
+have more than *one* ``ReservedParking``. Together the constraint and
+the qualifier specify a *one-to-one* relationship.
+
+Although the link is specified only on one of the objects, the
+relationship involves both of them and so it can be accessed from
+either end. To get the assigned ``ReservedParking`` given an
+``Employee`` the following query can be used:
+
+.. code-block:: edgeql
+
+    WITH Alice := (
+        SELECT Employee FILTER .name = 'Alice'
+    )
+    SELECT Alice.parking {
+        number
+    };
+
+The reverse lookup of who owns a particular ``ReservedParking`` spot
+can be done by using a backward link traversal like so:
+
+.. code-block:: edgeql
+
+    WITH Spot := (
+        SELECT ReservedParking FILTER .number = 42
+    )
+    SELECT Spot.<parking[IS Employee] {
+        name
+    };
+
+Backward link traversal requires to specify the original link's source
+type, but other than that it works the same way as forward traversal.
+
+
+Many-to-Many
+------------
+
+A *many-to-many* relationship represents the most generic kind of
+relationship without any exclusivity. For example, ``Person`` and
+``Movie`` in the following schema:
+
+.. code-block:: sdl
+
+    type Person {
+        required property name -> str {
+            constraint exclusive;
+        }
+        multi link likes -> Movie;
+    }
+    type Movie {
+        required property title -> str {
+            constraint exclusive;
+        }
+    }
+
+A ``Person`` can like multiple movies and each ``Movie`` can be liked
+by multiple people, thus making ``likes`` a *many-to-many*
+relationship. This type of relationship has the same symmetry as a
+*one-to-one* w.r.t. link traversal forward and backward, except that
+potentially multiple objects can reached in either direction. Here's
+the query for getting every ``Movie`` a given ``Person`` likes:
+
+.. code-block:: edgeql
+
+    WITH Cameron := (
+        SELECT Person FILTER .name = 'Cameron'
+    )
+    SELECT Cameron.likes {
+        title
+    };
+
+The reverse lookup of who likes a particular ``Movie``:
+
+.. code-block:: edgeql
+
+    WITH M := (
+        SELECT Movie FILTER .title = "Matrix"
+    )
+    SELECT M.<likes[IS Person] {
+        name
+    };
+
```

Comment:
Somewhere here is a good place to introduce link properties?

Reply:
Link properties are a messy topic all of their own. We have a few places scattered across the docs touching on them, but perhaps they should be reorganized.